client/components/webpack-build-monitor/index.jsx

```diff
@@ -5,148 +5,118 @@
  * External dependencies
  */
 import React from 'react';
-import { createStore } from 'redux';
 import classNames from 'classnames';
-import { find, includes, startsWith, identity } from 'lodash';
+import { find, startsWith } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import Spinner from 'components/spinner';
-import { combineReducers } from 'state/utils';
-
-// Action dispatched by wrapped console
-const CONSOLE_MESSAGE = 'CONSOLE_MESSAGE';
-
-const CONNECTED = 'CONNECTED';
-const DISCONNECTED = 'DISCONNECTED';
-const BUILDING_CSS = 'BUILDING_CSS';
-const BUILDING_JS = 'BUILDING_JS';
-const BUILDING_BOTH = 'BUILDING_BOTH';
-const ERROR = 'ERROR';
-const IDLE = 'IDLE';
-
-const doneBuilding = typeDone => state => {
-	if ( state === ERROR ) {
-		return state;
-	} else if ( state === BUILDING_BOTH ) {
-		return typeDone === BUILDING_JS ? BUILDING_CSS : BUILDING_JS;
-	}
-	return IDLE;
-};
-
-const buildJs = state => ( state === BUILDING_CSS ? BUILDING_BOTH : BUILDING_JS );
-const buildCss = state => ( state === BUILDING_JS ? BUILDING_BOTH : BUILDING_CSS );
 
 const MESSAGE_STATUS_MAP = {
-	'[HMR] connected': () => CONNECTED,
-	'[WDS] Disconnected!': () => DISCONNECTED,
-	'[HMR] bundle rebuilding': state => buildJs( state ),
-	'Building CSS…': state => buildCss( state ),
-	'CSS build failed': () => ERROR,
-	'[WDS] Nothing changed.': doneBuilding( BUILDING_JS ),
-	'Reloading CSS: ': doneBuilding( BUILDING_CSS ),
-	'[HMR] bundle rebuilt in': doneBuilding( BUILDING_JS ),
+	'[HMR] connected': { isConnected: true },
+	'[WDS] Disconnected!': { isConnected: false },
+	'[HMR] bundle rebuilding': { isBuildingJs: true },
+	'[HMR] Cannot find update (Full reload needed)': { needsReload: true },
+	"[HMR] The following modules couldn't be hot updated": { needsReload: true },
+	'Building CSS…': { isBuildingCss: true },
+	'CSS build failed': { hasError: true },
+	'[WDS] Nothing changed.': { isBuildingJs: false },
+	'Reloading CSS: ': { isBuildingCss: false },
+	'[HMR] bundle rebuilt in': { isBuildingJs: false },
 };
 
-const buildStatusReducer = ( state = IDLE, { type, message } ) => {
-	if ( type !== CONSOLE_MESSAGE ) {
-		return state;
-	}
+let buildState = {
+	hasError: false,
+	isBuildingCss: false,
+	isBuildingJs: false,
+	isConnected: true,
+	needsReload: false,
+};
+
+const interceptConsole = ( consoleObject, updater ) => {
+	[ 'error', 'log', 'warn' ].forEach( method => {
+		const unwrapped = consoleObject[ method ];
```

Comment:
may need to either bind this to `consoleObject` or invoke `unwrapped` with `apply` to set the correct `this`. I seem to remember IE getting angry when using console methods as free functions. 

IE: `const unwrapped = consoleObject[method].bind( console );`